third_party/amd/lib/TritonAMDGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -1348,23 +1352,47 @@ struct AtomicRMWOpConversion
       Value rmwMask = llMask ? b.and_(mask, maskElements[i]) : mask;
 
       Value operand;
+      Value rightNeighbourAddr;
+      Value enablePackedOpt;
       if (useDppForPackedF16) {
+        // First check if odd threads hold adjacent ptrs to even.
+        Value castedAddr = b.ptrtoint(i64_ty, rmwPtr);
+        Value leftNeighbourAddr =
+            genI32TiledOp(rewriter, shiftRightI32ByDpp, castedAddr);
+        Value isOddAndHoldsAdjacent = b.and_(
+            b.icmp_eq(b.urem(tid, b.i32_val(2)), b.i32_val(1)),
+            b.icmp_eq(
+                castedAddr,
+                b.add(leftNeighbourAddr,
+                      b.i64_val(valueElemTy.getIntOrFloatBitWidth() / 8))));
```

Comment:
Can we use some local variables with a proper name to make it more readable? It also avoid non deterministic evaluation order for C++ function calls.